File: OCH_Tao/ABC/B/278.js
let [H, M] = require("fs").readFileSync("/dev/stdin", "utf8").trim().split(" ").map(x => parseInt(x, 10));
while (true) {
  const A = Math.floor(H / 10);
  const B = H % 10;
  const C = Math.floor(M / 10);
  const D = M % 10;
  if (A * 10 + C < 24 && B * 10 + D < 60) {
    console.log(H, M);
    break;
  } else {
    M++;
    if (M === 60) {
      M = 0;
      H++;
      if (H === 24) {
        H = 0;
      }
    }
  }
}

File: OCH_Tao/ABC/B/278.py
H,M = map(int,input().split())
while True:
  A = H//10
  B = H%10
  C = M//10
  D = M%10
  if A*10+C<24 and B*10+D<60:
    print(H,M)
    break
  else:
    M+=1
    if M==60:
      M=0
      H+=1
      if H==24:
        H=0